### PR TITLE
Replace '.sdfx' directory references with core enum in replay debugger package

### PR DESCRIPTION
**Pull request by klewis-sfdc**
### What does this PR do?
To prevent potential disruption as the internals of the .sf and .sfdx evolve, use the enums from core instead of using the string reference to the state folder.

### What issues does this PR fix or reference?
@W-11633586@

### Functionality Before
The state folder was referenced via string value ".sfdx"

### Functionality After
The state folder is referenced via imported STATE_FOLDER enum from core: https://github.com/forcedotcom/sfdx-core/blob/a6e10b250ed8b4c23ec13f2c7f15ef218bd52135/src/global.ts#L33

Note: PR order: apex <-- lwc <-- replay-debugger

### CODE REVIEW

**Review by gbockus-sf — APPROVED** [review]

File: packages/salesforcedx-vscode-apex-replay-debugger/src/index.ts [hunk 1]
```diff
@@ -254,9 +257,7 @@ function getDialogStartingPath(): vscode.Uri | undefined {
       return vscode.Uri.file(sfdxCommandLogDir);
     }
     // If all else fails, fallback to the .sfdx directory in the workspace
-    return vscode.Uri.file(
-      path.join(vscode.workspace.workspaceFolders[0].uri.fsPath, '.sfdx')
-    );
+    return vscode.Uri.file(projectPaths.stateFolder());
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-apex-replay-debugger/src/index.ts:260]: was this already stubbed in the test or is there no test or ?

Comment (klewis-sfdc) [reply packages/salesforcedx-vscode-apex-replay-debugger/src/index.ts:260]: I don't think there was a test for this. Can I write a jest test for this function now?

Comment (gbockus-sf) [reply packages/salesforcedx-vscode-apex-replay-debugger/src/index.ts:260]: for sure 👍

File: packages/salesforcedx-vscode-apex-replay-debugger/test/unit/activation/getDialogStartingPath.test.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+describe('getDialogStartingPath', () => {
+  let hasRootWorkspaceStub: jest.SpyInstance;
+  let getLastOpenedLogFolderMock: jest.SpyInstance;
+  let folderExistsMock: jest.SpyInstance;
+  let getUriForMock: jest.SpyInstance;
+  const testPath = '/here/is/a/fake/path/to/';
+
+  beforeEach(() => {
+    hasRootWorkspaceStub = jest.spyOn(workspaceUtils, 'hasRootWorkspace');
+    getLastOpenedLogFolderMock = jest.spyOn(
```
Comment (klewis-sfdc) [inline packages/salesforcedx-vscode-apex-replay-debugger/test/unit/activation/getDialogStartingPath.test.ts:15]: Still trying to get these supporting functions from the dialogStartingPath module to be mocked correctly.  These don't seem to be working as written here.

File: packages/salesforcedx-vscode-apex-replay-debugger/test/unit/activation/getDialogStartingPath.test.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    expect(vsCodeUriMock).toHaveBeenCalledWith(fakePathToDebugLogsFolder);
+    expect((dialogStartingPathUri as vscode.Uri).path).toEqual(
+      fakePathToDebugLogsFolder
+    );
+  });
+
+  it('Should return state folder as fallback when project log folder not present', async () => {
+    hasRootWorkspaceStub.mockReturnValue(true);
+    const mockGet = jest.fn().mockReturnValue(undefined);
+    const mockExtensionContext: any = { workspaceState: { get: mockGet } };
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-apex-replay-debugger/test/unit/activation/getDialogStartingPath.test.ts:74]: total nit: but any time something exists in all tests it should probably be pulled up to the beforeEach.

File: packages/salesforcedx-vscode-apex-replay-debugger/test/unit/activation/getDialogStartingPath.test.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { LAST_OPENED_LOG_FOLDER_KEY } from '@salesforce/salesforcedx-apex-replay-debugger/src';
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-apex-replay-debugger/test/unit/activation/getDialogStartingPath.test.ts:8]: This file should be in the `test/jest` directory vs `test/unit`. 
`test/unit` -> old school
`test/jest` -> new school

**Review by gbockus-sf — CHANGES_REQUESTED** [review]
QE Complete: 
Only thing missing is adding the test:unit npm script 
* Verify that it can successfully open the last log path ✅
* Verify replay debugger load and initializes corrected ✅
* Verify it defaults to the .sfdx directory if not used previously ✅
* Verify test:unit script for vscode-apex-replay-debugger ❌
    * Need to add the test:unit npm script     "test:unit": "jest --coverage",

File: packages/salesforcedx-vscode-apex-replay-debugger/test/vscode-integration/commands/quickLaunch.test.ts [hunk 5]
```diff
@@ -362,7 +363,7 @@ describe('Quick launch apex tests', () => {
     expect(notificationServiceStub.called).to.equal(true);
     const notificationArgs = notificationServiceStub.getCall(0).args;
     expect(notificationArgs[0]).to.equal(
-      "Cannot read property 'length' of undefined"
+      "Cannot read properties of undefined (reading 'length')"
```
Comment (klewis-sfdc) [inline packages/salesforcedx-vscode-apex-replay-debugger/test/vscode-integration/commands/quickLaunch.test.ts:366]: Swapped to use the version of the error msg that was being reported from the failing test. Seems to be same error, just different string structure

**Review by gbockus-sf — APPROVED** [review]
* Verify that it can successfully open the last log path ✅
* Verify replay debugger load and initializes corrected ✅
* Verify it defaults to the .sfdx directory if not used previously ✅
* Verify test:unit script for vscode-apex-replay-debugger ✅
    * Need to add the test:unit npm script     "test:unit": "jest --coverage", ✅


* Verify unit test execution in circle ✅

File: packages/salesforcedx-vscode-apex-replay-debugger/test/vscode-integration/commands/quickLaunch.test.ts [hunk 6]
```diff
@@ -361,9 +362,14 @@ describe('Quick launch apex tests', () => {
 
     expect(notificationServiceStub.called).to.equal(true);
     const notificationArgs = notificationServiceStub.getCall(0).args;
-    expect(notificationArgs[0]).to.equal(
-      "Cannot read property 'length' of undefined"
+    // Seems that there are two different error msgs:
+    // On Windows: "Cannot read property 'length' of undefined"
+    // On Mac: "Cannot read properties of undefined (reading 'length')"
+    expect(notificationArgs[0].startsWith('Cannot read propert')).to.equal(
```
Comment (klewis-sfdc) [inline packages/salesforcedx-vscode-apex-replay-debugger/test/vscode-integration/commands/quickLaunch.test.ts:368]: @gbockus-sf FYI this test passes locally and fails the circle build due to the msg being different.  I tried to make the test work for both, but let me know if you think a different approach would be better here.

Comment (gbockus-sf) [reply packages/salesforcedx-vscode-apex-replay-debugger/test/vscode-integration/commands/quickLaunch.test.ts:368]: seems fine.  weird that it's different between mac & windows.

**Pull request merged**